src/components/structures/RoomView.js

```diff
@@ -714,7 +714,9 @@ module.exports = React.createClass({
         // NB. This unfortunately does not re-use the ChangeDisplayName component because
         // it doesn't behave quite as desired here (we want an input field here rather than
         // content-editable, and we want a default).
-        if (MatrixClientPeg.get().getRooms().length == 0) {
+        if (cli.getRooms().filter((r) => {
+            return !!r.getMember(cli.credentials.userId);
```

Comment:
To be fair, if they've joined a room and left it, that's probably good enough too, but ecplicitly looking for joins is probably clearer.